fastlane_bot/helpers/routehandler.py

```diff
@@ -320,7 +320,7 @@ def handle_custom_data_extras(self, platform_id: int, custom_data: bytes, exchan
 
         assert custom_data in "0x", f"[routehandler.py handle_uni_v3_router_switch] Expected the custom data field to contain '0x', but it contained {custom_data}. This function may need to be updated."
         if platform_id == self.ConfigObj.network.EXCHANGE_IDS.get(self.ConfigObj.network.UNISWAP_V3_NAME):
-            if self.ConfigObj.network.NETWORK in "ethereum" or exchange_name in self.ConfigObj.network.PANCAKESWAP_V3_NAME:
+            if self.ConfigObj.network.NETWORK in "ethereum" or exchange_name in [self.ConfigObj.network.PANCAKESWAP_V3_NAME, self.ConfigObj.network.BUTTER_V3_NAME, self.ConfigObj.network.AGNI_V3_NAME, self.ConfigObj.CLEOPATRA_V3_NAME]:
```

Comment:
The logic in this function is really obfuscated for no reason.

For one, the first `if` can generally be removed altogether.

Another thing is that the assertion statement is baffling for several different reasons, such as:
- The condition being tested is whether or not the string `"0x"` contains the value of `custom_data`, which doesn't make any sense, and is probably only true because the value of `custom_data` itself is `"0x"` (assuming that it is never `"0"` neither `"x"`)
- The error-message says the exact opposite of that condition, namely, that the value of `custom_data` should contain the string `"0x"`

Most likely, there is no reason to assert anything other than `custom_data == "0x"`, which should be done only when the input value of `custom_data` is overridden in each one of the `if` statements which follow that assertion.

In short, the code in this function should be something like:
```
if <the condition in the first 'if' statement>:
    assert custom_data == "0x"
    return '0x0000000000000000000000000000000000000000000000000000000000000000'
if <the condition in the second 'if' statement>:
    assert custom_data == "0x"
    return '0x0100000000000000000000000000000000000000000000000000000000000000'
return custom_data
```
I think that adding more logic instead of simplifying it is counterproductive.

Reply:
I completely agree with your assessment & have opened an issue to address this upstream so that the logic here can be more straightforward: https://github.com/bancorprotocol/fastlane-bot/issues/440

However, in the interest of time, does this need to be refactored immediately?

Reply:
No, I think that it is 99.99% safe; so given the linked issue, resolving.